### DISCUSSION TITLE
KindleForPC change

### DISCUSSION THREAD

**Discussion by Twombs** [noDRM/DeDRM_tools#263] [General]
I am currently using KindleForPC 1.26 on Windows 7 (32 Bit), my main web and download PC. Obviously I also run an older version of calibre on this PC.

Every few weeks I run KindleForPC to get my latest ebook purchases from Amazon, which I then strip the DRM from and convert to other formats, for backup purposes.

Today, two of my latest ebook purchases gave an error message, such that I was required to update KindleForPC to download them. The popup dialog mentioned a version for Windows 8 and newer. No idea whether the two ebooks require something in newer Windows (Font type perhaps) or whether the newer version of KindleForPC needed just doesn't run on Windows 7 now, and perhaps hasn't for a while. All the other ebooks downloaded to KindleForPC without issue, several of them, so the failure was limited to just two ebooks.

Ironically both the ebooks are DRM-Free at Kobo and can be downloaded directly.

They are also free at Smashwords, so I have no problem getting a DRM-Free copy of each ebook, so this is really just an alert for those who might appreciate it or have something to offer or suggest or discuss ... an interesting development or occurrence in any case.

Feel free to contribute something related. :)

**Comment by ElleKayEm** [comment]
Were the books published after January 3?  Amazon is requiring those to be downloaded with the latest Kindle for PC/Mac.  Then most books are delivered in KFX format which always has DRM.  (And the lastest KFX DRM cannot be removed.)  For now only way to get the older format in that case is having an E-ink Kindle.  See this post: https://www.mobileread.com/forums/showthread.php?t=283371

**Comment by Twombs** [comment]
Yes the books probably were, so you are probably right about that. Those two had only just been released, don't know about the others I got at the same time, as I often get a lot of free ebooks from Amazon via BookBub, at least two a day on average, and none of them gave issue, but then maybe they were all published before January 3. **EDIT** - Checking those 20 or so other ebooks, it appears they were all published before January 3.

Yep, I have already checked that thread out, first thing I did, and I tried one of the options but with no success. I may yet try another one ... download and side-load to my device, and strip the DRM, which I have never bothered with before, always done it with the KFX files in the Kindle For PC app (v1.26). I'll have a bit more of a play, because if more ebooks I grab are gonna require latest version of K4PC, then I need to seriously stop buying from Amazon, until removing DRM is solved. Luckily in this instance I could get those two ebooks free elsewhere and DRM-Free, all legit.

**Reply by ElleKayEm** [reply]
Don't expect much on KFX DRM removal.  Last 2 times DeDRM was updated for KFX, Amazon broke it by the next day.  They didn't need to push an update.  They just started sending differently encrypted books which the Kindle software/apps could already handle but DeDRM could not.

**Reply by Twombs** [reply]
Not really been an issue for me until now. I stayed on K4PC v1.17 for as long as I could then went to v1.19 (I think), then went to v1.26, which I have been on for ages. I did all I needed to, file renaming etc. I am going to have a go at toggling Registry write permissions, to see if I can get the AZW3 to download instead of KFX, but if that fails I guess I will need to try the device approach for any new ebook releases.

**Reply by ElleKayEm** [reply]
I'm still hanging in here with Kindle for PC 1.17 for now.  I do have an E-ink Kindle if I decide to buy a new Kindle book.  But I mostly shop at Kobo.

**Reply by Twombs** [reply]
Yep, Amazon seem determined to turn discerning customers away to other stores for ebooks.
Just a real shame some authors only provide to Amazon, and prices are not always on parity with the likes of Kobo, being cheaper at Amazon.

**Comment by romad275** [comment]
I’ve used Kindle for Mac 1.17 for years on my iMac without a problem until this latest bit of Amazon greed. Fortunately back in 2017 I bought a Kindle Paperwhite “just in case”. Well, the other day the “case” arrived so I found the unopened box, copied the DSN off of it into the DeDRM plugin, and it worked in Calibre just fine.

Sincerely,
Dennis B. Swaney

Cogito Ergo Mac

**Reply by Twombs** [reply]
I have an older Paperwhite, so I will likely go that route too now, for ebooks published after January 3.
Bit by bit, Amazon have been turning me away from buying ebooks from them, and this is likely shaping up to being the final nail in the coffin. It may even mean I stop buying from a few authors that only publish Kindle ebooks. Hell, I have a huge backlog anyway, so I can even wait out things for a good while.

It started with the missing covers fiasco, then the forced secret update fiasco, then the missing title and author names in emails that aren't receipts but are fiasco, then DRM-Free ebooks that aren't due to KFX fiasco, web page bloat, etc.

**Reply by Twombs** [reply]
Well I confirmed that when I downloaded the two ebooks for the copy via USB option, that they were AZW3 not KFX, and that they are DRM-Free, so were viewable in calibre without the DeDRM plugin being installed. I'd never needed to confirm that before, so now I know that you can get DRM-Free listed ebooks via that method from Amazon.

The downside, is I now need to wait for a pre-order which will undoubtedly have DRM, and test that with plugin and ID ... unless I get a new free Kindle ebook meanwhile which has DRM. I won't cancel my pre-order as it is part of a series and only available at Amazon. It could however be my last new purchase from Amazon.

**Comment by Twombs** [comment]
Well, that was less fuss than I thought it would be. No doubt, because I'd never really needed to think about it until now, I had this idea I needed to actually connect my E-Ink Kindle for this process. Thinking about it today, and going through the motions, it then seemed logical not to need my E-Ink Kindle, other than to turn it on and grab the serial number from the About section in settings. The only slight issue was when I entered the serial number into the DeDRM plugin in calibre, and got a complaint about too many characters, which I quickly realized was due to the spaces, which once removed, was correct. Adding the DRM ebook to calibre, then worked well, with DRM removed in the process. Luckily I now only need to do this for Kindle ebooks published from the 3rd January 2023 onward, but it ain't too onerous a process in any case. The only likely fly in the ointment for me, is that I don't leave the DeDRM plugin installed, so I imagine I will have to keep re-entering the serial number when required, unless I can figure a way to do that programmatically.
So anyway, it seems I am future proofed for now, and can still purchase ebooks at Amazon when I need to.

**Reply by ElleKayEm** [reply]
I find copy and paste of the serial number from the website is the easiest way to get it in there.  Although the settings are stored in a json file in calibre's configuration folder, and it remains even when DeDRM is uninstalled.  So when you reinstall DeDRM, you'll be good to go.  But why uninstall the DeDRM plugin?  You could disable it when you don't want it.

**Reply by Twombs** [reply]
Thanks, good to know. I'll check out that json file.
I have always loaded it and enabled it and removed it with batch file code, just in case something goes looking for it within calibre. What can I say, I am a cautious bloke. I also mostly add and edit in calibre using code anyway, so no biggie to only have the plugin present when I need it.
